### ISSUE TITLE
Verification of string arguments in datasets?

### ISSUE THREAD

**Issue by pmeier** [pytorch/vision#1132]
In the spirit of standardizing the datasets (#1080), I think it would be a good idea to have something like a `verify_str_arg()` function that checks whether if a given argument is a `str` (or a valid subclass) and if hat has a valid value. Right now we have a multitude of slightly different variants:

https://github.com/pytorch/vision/blob/bbd363ca2713fb68e1e190206578e600a87baf90/torchvision/datasets/celeba.py#L76-L86

https://github.com/pytorch/vision/blob/bbd363ca2713fb68e1e190206578e600a87baf90/torchvision/datasets/sbd.py#L68-L69

https://github.com/pytorch/vision/blob/bbd363ca2713fb68e1e190206578e600a87baf90/torchvision/datasets/voc.py#L103-L106

Is this something we want to do?

**Comment by fmassa** [comment]
I think the split names depend on the dataset. For example, `train` / `val` / `valminusminival` for some version of COCO.

Some consistency might be good, but not at the expense of generality.

**Comment by pmeier** [comment]
My mistake for choosing these examples with varying argument names, but that is not what I meant. I don't want to generalize the values of the `split` or similar arguments. What I propose is a function (probably in `datasets.utils`) that handles the verification:

```python
def verify_str_arg(value, valid_values, arg):
    if not isinstance(value, str):
        raise ...
    if value not in valid_values:
        raise ...
    return value
```

With that all error messages would be the same across datasets.

**Comment by fmassa** [comment]
Oh, that's definitely something that we would want to have! I completely misunderstood your point.

**Comment by pmeier** [comment]
Closed by #1167
